src/GeniusYield/TxBuilder.hs

```diff
@@ -93,3 +95,138 @@ getTxBalance pkh tx = do
     f utxo
         | isRelevantAddress $ utxoAddress utxo = utxoValue utxo
         | otherwise                            = mempty
+
+-------------------------------------------------------------------------------
+-- Transaction builder
+-------------------------------------------------------------------------------
+
```

Comment:
The code below hasn't been finalized as I'm waiting on feedback around what type signature to have for the parallel and chaining functions. See related comments on PR.